syntax = "proto3";
import "google/protobuf/timestamp.proto";
// import "google/protobuf/any.proto";

package gos;

service Transaction {

    // Starts a new transaction
    rpc StartTransaction(StartTransactionRequest) returns (TxnId) {}
    rpc GetOidState(GetOidTxn) returns (State) {}
    rpc SaveDublinCore(DublinCore) returns (TxnId) {}
}


message StartTransactionRequest {
    bool write = 1;
    string user = 2;
    //repeated string roles = 3;
    string path = 4;
}

message TxnId {
    uint64 tid = 1;
    uint32 part = 2;
}

message GetOidTxn {
    uint64 tid = 1;
    string oid = 2;
    bool update = 3;
}

message State {
    bytes state = 1;
    string of = 2;
    string parent_id = 3;
    string id = 4;
    string type = 5;
    string otid = 6;
}

message DublinCore {
  string tags = 1;
  google.protobuf.Timestamp creation_date = 2;
  google.protobuf.Timestamp effective_date = 3;
  google.protobuf.Timestamp expiration_date = 4;
  string creators = 5;
}
